### PR TITLE
Fix PipelineTest, SharedStudyTest, and PanoramaPublicTests failures

### PR DESCRIPTION
**Pull request by Sigmonia**
#### Rationale
Fix failing tests

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/2663

#### Changes
* Prevent setting updating logFile after Job completion for non-cloud jobs (For cloud jobs: log is moved to cloud working dir)
* Fix Job LocalDirectory overwriting log file location for split-able jobs causing error
* Fix incorrect expected value (StudyTest

### CODE REVIEW

File: study/test/src/org/labkey/test/tests/study/SharedStudyTest.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
     private static final String STUDY1 = "Study001";
     private static final String STUDY2 = "Study002";
     private static final String SHARED_DEMOGRAPHICS = "P_One_Shared";
-    private static final String SHARED_DEMOGRAPHICS_ID = "5001";
+    private static final String SHARED_DEMOGRAPHICS_ID = "10001";
```
Comment (labkey-jeckels) [inline study/test/src/org/labkey/test/tests/study/SharedStudyTest.java:62]: This seems unrelated.

**Review by labkey-jeckels — APPROVED** [review]

File: study/test/src/org/labkey/test/tests/study/SharedStudyTest.java (hunk 1 above)
Comment (Sigmonia) [reply study/test/src/org/labkey/test/tests/study/SharedStudyTest.java:62]: Agreed, and I cant find the originating change. But it was needed to unblock the SharedStudyTest's setup method.

**Comment by Sigmonia** [comment]
https://teamcity.labkey.org/buildConfiguration/LabkeyTrunk_ModuleSuites_MacCossLabModulesPostgres/1588748?buildTab=overview

File: study/test/src/org/labkey/test/tests/study/SharedStudyTest.java (hunk 1 above)
Comment (labkey-adam) [reply study/test/src/org/labkey/test/tests/study/SharedStudyTest.java:62]: I'll bet it was related to the dataset ID collisions change: https://github.com/LabKey/platform/pull/2707
